keeper/keeper.go

```diff
@@ -135,8 +135,16 @@ func (k *Keeper) SendRestrictionFn(ctx context.Context, _, toAddr sdk.AccAddress
 	}
 
 	mintingDenom := k.ftfKeeper.GetMintingDenom(ctx).Denom
-	if coins.AmountOf(mintingDenom).IsPositive() {
-		err = k.PendingTransfers.Set(ctx, account.Address, *account)
+	if len(coins) != 1 || !coins.AmountOf(mintingDenom).IsPositive() {
+		return toAddr, fmt.Errorf("autocctp accounts can only receive %s coins", mintingDenom)
+	}
+
+	if err = k.PendingTransfers.Set(ctx, account.Address, *account); err != nil {
+		k.logger.Error(`'while setting account for pending transfers in send restriction`,
+			"account", account.Address,
+			"amount", coins.AmountOf(mintingDenom).String(),
+			"error", err,
+		)
 	}
 
 	return toAddr, err
```

Comment:
We shouldn't return any error here right?

```suggestion
	return toAddr, nil
```